File: leetcode/python/hard/2444_count_subarrays_with_fixed_bounds.py

class Solution:
    def countSubarrays(self, nums: list[int], minK: int, maxK: int) -> int:
        res = 0
        mini = maxi = -1
        j = 0
        for i, n in enumerate(nums):
            if not minK <= n <= maxK:
                mini = maxi = -1
                j = i + 1
            if n == minK:
                mini = i
            if n == maxK:
                maxi = i

            res += max(0, min(mini, maxi) - j + 1)

        return res


s = Solution()
print(s.countSubarrays(nums = [3,1,5,2,0,7,5,2,1], minK = 1, maxK = 5))
print(s.countSubarrays(nums = [1,3,5,2,0,7,5], minK = 1, maxK = 5))
print(s.countSubarrays(nums = [1,1,1,1], minK = 1, maxK = 1))
print(s.countSubarrays([35054,398719,945315,945315,820417,945315,35054,945315,171832,945315,35054,109750,790964,441974,552913], 35054, 945315))

File: leetcode/rust/src/hard/2444_count_subarrays_with_fixed_bounds.rs
impl Solution {
    pub fn count_subarrays(nums: Vec<i32>, min_k: i32, max_k: i32) -> i64 {
        let mut res = 0;
        let mut left = 0;
        let mut min_index = -1;
        let mut max_index = -1;
        for (right, &num) in nums.iter().enumerate() {
            if num > max_k || num < min_k {
                left = right + 1;
                min_index = -1;
                max_index = -1;
            }
            if num == min_k {
                min_index = right as i64;
            }
            if num == max_k {
                max_index = right as i64;
            }
            res += 0.max(min_index.min(max_index) - left as i64 + 1);
        }
        res as i64
    }
}